wiji/task.py

```diff
@@ -670,6 +671,56 @@ def _get_task_options(self, *args: typing.Any, **kwargs: typing.Any) -> TaskOpti
         return task_options
 
 
+def task_decor(
+    the_broker: broker.BaseBroker, queue_name: str, bind: bool = False, **options
+) -> typing.Callable[..., Task]:
+    """
+    decorator function that creates a `wiji.task.Task` instance from a function.
+
+    Parameters:
+        the_broker: The broker to be used by the task
+        queue_name: The name of the queue where that task will be saved in the_broker
+        bind: Whether to provide access to `self` (of the task type instance).
+        options: all the other attributes that can be passed to a `wiji.task.Task`
+
+    Returns:
+        function decorator that can decorate another function that will return a `wiji.task.Task` class
+
+    Usage:
+        @task_decor(the_broker=InMemoryBroker(), queue_name="q1")
+        async def hello():
+            print("hello world.")
+
+        await hello().delay()
```

Comment:
This is weird. 
```python
@task_decor(the_broker=InMemoryBroker(), queue_name="q1")
async def hello():
    print("hello world.")
    await hello().delay()

await hello().delay()
```
calling a function like; `hello().delay()` instead of `hello.delay()`.   

this may lead to a lot of confusion/questions and weird error messages.  
the func decorator may not be worth it.